plugins/nf-schema/src/main/nextflow/validation/utils/Common.groovy

```diff
@@ -59,8 +59,19 @@ public class Common {
     //
     // Find a value in a nested map
     //
-    public static Object findDeep(Map m, String key) {
-        if (m.containsKey(key)) return m[key]
-        m.findResult { k, v -> v instanceof Map ? findDeep(v, key) : null }
+    public static Object findDeep(Object m, String key) {
+        if (m instanceof Map) {
+            if (m.containsKey(key)) {
+                return m[key]
+            } else {
+                return m.findResult { k, v -> findDeep(v, key) }
+            }
+        } 
+        else if (m instanceof List) {
+            def result = null
+            return m.findResult { element -> findDeep(element, key) }
+            return result
```

Comment:
result is never returned. Can `findResult` be `null`? If not, then we need a check to return `null`
```suggestion
            return m.findResult { element -> findDeep(element, key) }
```

Reply:
Thank you for spotting this! Must have been left in by accident